addons/project/static/src/project_task_kanban/project_task_kanban_renderer.js

```diff
@@ -1,14 +1,105 @@
 /** @odoo-module */
 
+import { useService } from '@web/core/utils/hooks';
 import { KanbanRenderer } from '@web/views/kanban/kanban_renderer';
+import { FormViewDialog } from "@web/views/view_dialogs/form_view_dialog";
+
+const { onWillStart } = owl;
 
 export class ProjectTaskKanbanRenderer extends KanbanRenderer {
+    setup() {
+        super.setup();
+        this.userService = useService('user');
+
+        this.isProjectManager = false;
+        onWillStart(async () => {
+            if (!this.isGroupedByPersonalStages) { // no need to check it if the group by is personal stages
+                this.isProjectManager = await this.userService.hasGroup('project.group_project_manager');
+            }
+        });
+    }
+
+    get canMoveRecords() {
+        let canMoveRecords = super.canMoveRecords;
+        if (!canMoveRecords && this.canResequenceRecords && this.isGroupedByPersonalStages) {
+            const { groupByField } = this.props.list;
+            const { modifiers } = groupByField;
+            canMoveRecords = !(modifiers && modifiers.readonly);
+        }
+        return canMoveRecords;
+    }
+
+    get canResequenceGroups() {
+        let canResequenceGroups = super.canResequenceGroups;
+        if (!canResequenceGroups && this.isGroupedByPersonalStages) {
+            const { modifiers } = this.props.list.groupByField;
+            const { groupsDraggable } = this.props.archInfo;
+            canResequenceGroups = groupsDraggable && !(modifiers && modifiers.readonly);
+        }
+        return canResequenceGroups;
+    }
+
+    get isGroupedByStage() {
+        if (!this.props.list.groupByField) {
+            return false;
+        }
+        return this.props.list.groupByField.name === 'stage_id';
```

Comment:
Nice, indeed, I will just convert `this.props.list.groupByField` to be sure to return a boolean value in that property instead of having an undefined value for instance. 🙂

Reply:
I will move the method in `DynamicGroupList` class of the `KanbanModel` since I need that method in that class and `this.props.list` is a instance of `DynamicGroupList` class so it should be good. 🙂